src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3636 @@
+            env.fund(XRP(10000), alice, bob);
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+
+            // AccountSet Txn
+            auto const noopTxn = env.jt(noop(alice), seq(aliceSeq + 1));
+            env(noopTxn, ter(terPRE_SEQ));
+
+            // Batch Txn
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const [txIDs, batchID] = submitBatch(
+                env,
+                tesSUCCESS,
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq + 2));
+            env.close();
+
+            {
+                std::vector<TestLedgerData> testCases = {
+                    {0, "Batch", "tesSUCCESS", batchID, std::nullopt},
+                    {1, "Payment", "tesSUCCESS", txIDs[0], batchID},
+                    {2, "Payment", "tesSUCCESS", txIDs[1], batchID},
+                };
+                validateClosedLedger(env, testCases);
+            }
+
+            env.close();
+            {
+                // next ledger is empty
+                std::vector<TestLedgerData> testCases = {};
+                validateClosedLedger(env, testCases);
+            }
+        }
+
+        // After Batch Txn w/ same sequence
+        {
+            // IMPORTANT: The batch txn is applied first, then the noop txn.
+            // Because of this ordering, the noop txn is not applied and is
+            // overwritten by the payment in the batch transaction.
+            test::jtx::Env env{*this, envconfig()};
+            env.fund(XRP(10000), alice, bob);
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const [txIDs, batchID] = submitBatch(
+                env,
+                tesSUCCESS,
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq + 2));
+
+            auto const noopTxn = env.jt(noop(alice), seq(aliceSeq + 1));
+            auto const noopTxnID = to_string(noopTxn.stx->getTransactionID());
+            env(noopTxn);
+            env.close();
+
+            {
+                std::vector<TestLedgerData> testCases = {
+                    {0, "Batch", "tesSUCCESS", batchID, std::nullopt},
+                    {1, "Payment", "tesSUCCESS", txIDs[0], batchID},
+                    {2, "Payment", "tesSUCCESS", txIDs[1], batchID},
+                };
+                validateClosedLedger(env, testCases);
+            }
+
+            env.close();
+            {
+                // next ledger is empty
+                std::vector<TestLedgerData> testCases = {};
+                validateClosedLedger(env, testCases);
+            }
+        }
+    }
```

Comment:
can we add another test where the batch txn results in `terPRE_SEQ` in the open ledger, but then it succeeds in the closed ledger?

Reply:
added: https://github.com/XRPLF/rippled/pull/5060/commits/ac54f33de3c05d205b803b2092e918ffc5094399